src/web/app/pages-instructor/instructor-course-student-edit-page/instructor-course-student-edit-page.component.ts

```diff
@@ -9,16 +30,144 @@ import { ActivatedRoute } from '@angular/router';
+    (this.emailFieldSubscription as Subscription).unsubscribe();
+    (this.teamFieldSubscription as Subscription).unsubscribe();
+  }
+
+  /**
+   * Loads student details required for this page.
+   */
+  loadStudentEditDetails(courseid: string, studentemail: string): void {
+    const paramsMap: { [key: string]: string } = { courseid, studentemail };
+    this.httpRequestService.get('/students/editDetails', paramsMap)
+        .subscribe((resp: StudentEditDetails) => {
+          this.student = resp.student;
+          if (!this.student) {
+            this.statusMessageService.showErrorMessage('Error retrieving student details');
+          } else {
+            this.studentemail = this.student.email;
+            this.isOpenOrPublishedEmailSentForTheCourse =
+                resp.isOpenOrPublishedEmailSentForTheCourse;
+            this.initEditForm();
+          }
+        }, (resp: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorMessage(resp.error.message);
+        });
+  }
+
+  /**
+   * Initializes the student details edit form with the fields fetched from the backend.
+   * Subscriptions are set up to listen to changes in the 'teamname' fields and 'newstudentemail' fields.
+   */
+  private initEditForm(): void {
+    this.editForm = new FormGroup({
+      studentname: new FormControl(this.student.name),
+      sectionname: new FormControl(this.student.section),
+      teamname: new FormControl(this.student.team),
+      newstudentemail: new FormControl(this.studentemail), // original student email initialized
+      comments: new FormControl(this.student.comments),
+    });
+    this.teamFieldSubscription =
+        (this.editForm.get('teamname') as AbstractControl).valueChanges
+            .subscribe(() => {
+              this.isTeamnameFieldChanged = true;
+            });
+
+    this.emailFieldSubscription =
+        (this.editForm.get('newstudentemail') as AbstractControl).valueChanges
+            .subscribe(() => this.isEmailFieldChanged = true);
+  }
+
+  /**
+   * Handles logic related to showing the appropriate modal boxes
+   * upon submission of the form. Submits the form otherwise.
+   */
+  onSubmit(confirmDelModal: any, resendPastLinksModal: any): void {
+    if (this.isTeamnameFieldChanged) {
+      this.ngbModal.open(confirmDelModal);
+    } else if (this.isEmailFieldChanged) {
+      this.ngbModal.open(resendPastLinksModal);
+    } else {
+      this.submitEditForm();
+    }
+  }
+
+  /**
+   * Shows the `resendPastSessionLinks` modal if email field has changed.
+   * Submits the form  otherwise.
+   */
+  deleteExistingResponses(resendPastLinksModal: any): void {
+    if (this.isEmailFieldChanged) {
+      this.ngbModal.open(resendPastLinksModal);
+    } else {
+      this.submitEditForm();
+    }
+  }
+
+  /**
+   * Sets the boolean value of `isSessionSummarySendEmail` to true if
+   * user chooses to resend past session link to the new email.
+   * Submits the form otherwise.
```

Comment:
This is not accurate? It submits the form regardless of the value.

Reply:
@wkurniawan07 This is intended. The default value of `isSessionSummarySendEmail` is declared false initially. 

In the event the user clicks `No, just save the changes`, the form will still be submitted with the boolean being false. If the user clicks `Yes, save changes and resend links`, the form will be submitted with the boolean being true.

Reply:
So "Submits the form **otherwise**." is not accurate because it submits the form always.
Not a big deal since it's just a comment and can be addressed later.